### PR TITLE
SCIM: add docs

### PR DESCRIPTION
**Pull request by vdavid**
- Completes https://github.com/sourcegraph/sourcegraph/issues/49216
- Completes https://github.com/sourcegraph/sourcegraph/issues/49020

Three changes:

- Added Okta-specific setup info for admins
- Added dev docs to capture the most crucial info that we only had in the SCIM Google Doc
- Linked SCIM from SAML pages for Okta and Azure AD to improve discoverability at the points where it's the most relevant

## Test plan

Only docs changes.
I've clicked through the links.

### CODE REVIEW

File: doc/admin/scim.md [hunk 1]
```diff
@@ -43,6 +43,24 @@ To configure:
...
+To set up user provisioning in Okta, you must first set up a new app integration of the "SAML 2.0" type, then configure it to use SCIM. Here are the steps to do this:
+
+1. Follow our [SAML guide](auth/saml/okta) to set up a new app integration with SAML, then open the integration you just created.
+    - If you already have the integration, just open your existing app integration.
+2. Go to the "General" tab and click "Edit" in the "App Settings" section.
+3. Set "Provisioning" to "SCIM". This makes a new tab called "Provisioning" appear.
+4. Go to the new tab, and click "Edit"
+   - Set "SCIM connector base URL" to `{yourSourcegraphUrl}/.api/scim/v2`
+   - Set "Unique identifier field for users" to `userName`
+   - Check the first three items in `Supported provisioning actions`
```
Comment (chwarwick) [inline doc/admin/scim.md:57]: May be worth it to say what these items are incase Okta changes the order of them.

**Review by chwarwick — APPROVED** [review]

File: doc/dev/background-information/scim_api.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+SCIM is a REST API specification standard for provisioning and deprovisioning users. It can also handle group (de)provisioning, but we don't support that yet.
+
+If you're just getting started with SCIM, watch [this](https://www.youtube.com/watch?v=LBZPPBOdImU) (1.5min) and [this](https://www.youtube.com/watch?v=ID-ApdGu2Hc) (2.5 min) video.
+
+IETF RFCs [7642](https://www.rfc-editor.org/rfc/rfc7642), [7643](https://www.rfc-editor.org/rfc/rfc7643), and [7644](https://www.rfc-editor.org/rfc/rfc7644) are pretty dry, but very useful reference materials.
+
+Our implementation is in [this folder](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/enterprise/internal/scim).
+
+To try it, follow the "How to use" section of our [SCIM admin guide](doc/admin/scim.md), then see the [manual-testing-with-postman](#manual-testing-with-postman) section below to see it in action. It should take you 10–20 minutes to set up everything and see the full CRUD magic happen.
```
Comment (kopancek) [inline doc/dev/background-information/scim_api.md:11]: I think linking to the section itself might be more useful.

```suggestion
To try it, follow the [How to use section of our SCIM admin guide](doc/admin/scim.md#how-to-use), then see the [manual-testing-with-postman](#manual-testing-with-postman) section below to see it in action. It should take you 10–20 minutes to set up everything and see the full CRUD magic happen.
```

File: doc/dev/background-information/scim_api.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Using ngrok is a good idea because it lets you inspect the requests and responses at http://127.0.0.1:4040/inspect/http.
```
Comment (kopancek) [inline doc/dev/background-information/scim_api.md:13]: I think we can reword this a little. Also, maybe worth spending some effort on ngrok setup or at least linking some standard ngrok setup? Let's say I never used it and have no idea how I would set it up.

```suggestion
To inspect the requests and responses, you can use ngrok. You can then open your browser at http://127.0.0.1:4040/inspect/http to dive deep into the details.
```

File: doc/dev/background-information/scim_api.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- Download and run Postman
+- Go to `File | Import...` and import the JSON.
+- You can import it into Postman and run the requests from top to bottom.
+- Go to `SCIM | Variables` and set your token and optionally your SCIM base URL, in the format of `https://{sourcegraph URL}/.api/scim/v2`.
+- For a quick test, run "User create" and "User delete".
+- For a complete test, run all requests from top to bottom. They should all pass.
+
+## Validators
+
+We use three validators: two for Okta and one for Azure AD.
```
Comment (kopancek) [inline doc/dev/background-information/scim_api.md:29]: I would also want to understand why we use the validators, e.g. what are they useful for.

File: doc/dev/background-information/scim_api.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Okta SPEC test – Follow [this guide](https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#test-your-scim-api) to set it up in five minutes. Tests should be green.
+2. Okta CRUD test – Follow [this guide](https://developer.okta.com/docs/guides/scim-provisioning-integration-test/main/) to set up these tests in your Runscope. This also needs access to an Okta application, which you can find [here](https://dev-433675-admin.oktapreview.com/admin/app/dev-433675_k8ssgdevorgsamlscim_1/instance/0oa1l85zn9a0tgzKP0h8/). Log in with shared credentials in 1Password. Tests should be green, except for the last ones that rely on soft deletion which we don't support yet.
+3. Azure AD validator – It's [here](https://scimvalidator.microsoft.com). It doesn't have a lot of docs. Just use "Discover my schema", then enter the endpoint (for example, https://sourcegraph.test:3443/search/.api/scim/v2) and the token you have in your settings. It should work right away, and all tests should pass.
+
+
+## Publishing on Okta
+
+Our integration is [here](https://oinmanager.okta.com/app-integration/1731). To access this page, you will need a bit of a hackery:
```
Comment (kopancek) [inline doc/dev/background-information/scim_api.md:38]: Why are we publishing to Okta? What are we publishing to Okta? :)

File: doc/admin/auth/saml/azure_ad.md [hunk 6]
```diff
@@ -32,3 +32,5 @@
  ]
 }
 ```
+
+Optional: if you also need automatic provisioning, continue to our [SCIM docs](../../scim).
```
Comment (kopancek) [inline doc/admin/auth/saml/azure_ad.md:36]:
```suggestion
> NOTE: Optional, but recommended: [add automatic provisioning of users with SCIM](../../scim). 
```

File: doc/admin/scim.md [hunk 7]
```diff
@@ -43,6 +43,24 @@ To configure:
...
 
+## Setting up Okta as the IdP
+
+To set up user provisioning in Okta, you must first set up a new app integration of the "SAML 2.0" type, then configure it to use SCIM. Here are the steps to do this:
+
+1. Follow our [SAML guide](auth/saml/okta) to set up a new app integration with SAML, then open the integration you just created.
+    - If you already have the integration, just open your existing app integration.
+2. Go to the "General" tab and click "Edit" in the "App Settings" section.
+3. Set "Provisioning" to "SCIM". This makes a new tab called "Provisioning" appear.
+4. Go to the new tab, and click "Edit"
```
[the remark this answers is not in the document]
Comment (vdavid) [reply doc/admin/scim.md:54]: Ah, okay. My IDE does this automatically, but I agree that it's more accessible for others if we just use 1s

File: doc/dev/background-information/scim_api.md (hunk 5 above)
Comment (vdavid) [reply doc/dev/background-information/scim_api.md:38]: The app integration. When I tested SCIM with @milespatton, he pointed out that having an app in the Okta marketplace makes setting up SAML and SCIM very convenient from a tech-ops POV. When we buy new software, Miles kind of expects to see it in the Okta marketplace, and just do a next-next-finish sort of integration with our IdP (Okta).

I admit that adding this is slightly out of scope for the SCIM project, but it had the benefit of allowing us to run an extra set of validators with Okta and helped us catch bugs, so it was already beneficial. But also, based on what Miles said, it seems extremely helpful to admins if we had our stuff in there.

File: doc/admin/scim.md [hunk 8]
```diff
@@ -43,6 +43,24 @@ To configure:
    https://sourcegraph.company.com/.api/scim/v2/Users
    ```
 
+## Setting up Okta as the IdP
+
+To set up user provisioning in Okta, you must first set up a new app integration of the "SAML 2.0" type, then configure it to use SCIM. Here are the steps to do this:
```
[the remark this answers is not in the document]
Comment (vdavid) [reply doc/admin/scim.md:48]: I think people who never heard of Okta won't read/follow this section anyway. But your link might be useful both for users who know and who don't know Okta, so I'll add it. Thanks!

File: doc/admin/scim.md [hunk 9]
```diff
@@ -43,6 +43,24 @@ To configure:
...
+3. Set "Provisioning" to "SCIM". This makes a new tab called "Provisioning" appear.
+4. Go to the new tab, and click "Edit"
+   - Set "SCIM connector base URL" to `{yourSourcegraphUrl}/.api/scim/v2`
+   - Set "Unique identifier field for users" to `userName`
+   - Check the first three items in `Supported provisioning actions`
+   - Set "Authentication mode" to "HTTP Header"
+   - Under "HTTP Header", paste the same alphanumeric bearer token you used in your site config.
+   - Click "Save".
+
+Note that you can also use our [SAML](auth/saml/okta) and [OpenID Connect](auth#openid-connect) integrations with Okta.
```
[the remark this answers is not in the document]
Comment (vdavid) [reply doc/admin/scim.md:62]: When I sat down with @milespatton to set up SCIM with our Okta, we didn't manage to get a "Provisioning" tab if we chose anything other than "SAML 2.0" on this screen:

CleanShot 2023-03-21 at 13 24 38@2x

So I guess that's the safe bet. I'm not sure how to make OIDC work at the same time—probably as an additional application.

[the remark this answers is not in the document]
Comment (vdavid) [reply doc/admin/scim.md:62]: I've reformatted the note.

**Pull request merged**